cmd/kubeadm/app/cmd/token.go

```diff
@@ -253,19 +257,46 @@ func RunCreateToken(out io.Writer, client clientset.Interface, cfgPath string, i
 		return err
 	}
 
+	if setupForMasterJoin {
+		klog.V(1).Infoln("[token] calling certificate creation")
+		// Create new certificate
+		internalcfg.CertificateKey, err = copycerts.CreateCertificateKey()
+		if err != nil {
+			return err
+		}
+
+		if err := copycerts.UploadCerts(client, internalcfg, internalcfg.CertificateKey); err != nil {
```

Comment:
i agree with @ereslibre that this is problematic. ideally we should not create and new key and upload the certs from the token command.